core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -350,19 +342,152 @@ describe('Contributor dashboard Admin page', () => {
       expect(component.selectedLanguage.id).toBe(nonDefaultLanguage.id);
     }));
 
-    it('should select last activity from dropdown', fakeAsync(() => {
+    it('should set default first activity at 0 days', fakeAsync(() => {
       component.ngOnInit();
       tick();
       fixture.detectChanges();
 
-      let lastActivityDays = 7;
-      component.selectLastActivity(lastActivityDays);
-      expect(component.selectedLastActivity).toEqual(lastActivityDays);
-      lastActivityDays = 0;
-      component.selectLastActivity(lastActivityDays);
-      expect(component.selectedLastActivity).toEqual(lastActivityDays);
+      expect(component.selectedFirstActivityInDays).toEqual(0);
     }));
 
+    it(
+      'should update first activity in days via dateChange' + 'event',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        const newDate: Date = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 7
+        );
+        component.changeFirstActivity(newDate);
+        expect(component.selectedFirstActivityInDays).toEqual(7);
+      })
+    );
+
+    it(
+      'should update first activity in days when selected first' +
+        'date is valid',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 7
+        );
+        component.selectFirstActivity();
+        expect(component.selectedFirstActivityInDays).toEqual(7);
+      })
+    );
+
+    it(
+      'should not update first activity in days when selected first' +
+        ' date is more than today',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() + 24 * 60 * 60 * 1000
+        );
+        component.selectFirstActivity();
+        expect(component.selectedFirstActivityInDays).toEqual(0);
+      })
+    );
+
+    it(
+      'should not update first activity in days when selected first' +
+        ' date is less than selected last date',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 91
+        );
+        component.selectFirstActivity();
+        expect(component.selectedFirstActivityInDays).toEqual(0);
+      })
+    );
```

Comment:
```ts
    it(
      'should not change filter when given date to start including ' +
        'contributions is after the date to stop including contributions',
      fakeAsync(() => {
        component.ngOnInit();
        tick();
        fixture.detectChanges();
        const initialFirstActivity = component.filter.firstActivity;
        const numberOfDaysSinceFirstActivity = (
          component.filter.lastActivity + 1
        );

        component.changeFirstActivity(new Date(
          new Date().getTime() - 24 * 60 * 60 * 1000 * numberOfDaysSinceFirstActivity
        )); // Reminder to refactor this Date creation by calling another method

        expect(
          component.filter.firstActivity
        ).toEqual(initialFirstActivity);
      })
    );
```

Reply:
Change test name to `does not change filter by users' last activity when given start date comes after filter's current end date`.

Reply:
done